src/carousel/carousel.ts

```diff
@@ -196,6 +208,48 @@ export class NgbCarousel implements AfterContentChecked,
     return isFirstSlide ? (this.wrap ? slideArr[slideArr.length - 1].id : slideArr[0].id) :
                           slideArr[currentSlideIdx - 1].id;
   }
+
+  private _getSlideEventDirection(currentActiveSlideId: string, nextActiveSlideId: string):
+      NgbSlideCarouselEventDirection {
+    const currentActiveSlideIdx = this._getSlideIdxById(currentActiveSlideId);
+    const nextActiveSlideIdx = this._getSlideIdxById(nextActiveSlideId);
+
+    return currentActiveSlideIdx > nextActiveSlideIdx ? NgbSlideCarouselEventDirection.RIGHT :
+                                                        NgbSlideCarouselEventDirection.LEFT;
+  }
+}
+
+/**
+* The payload of the slide event fired when the slide transition is completed
+*/
+export interface NgbSlideCarouselEvent {
+  /**
+   * Event type
+   */
+  type: 'slide';
```

Comment:
of course 2 different events and not only one discriminated by strings :) it's the bs event structure having the 'type' prop so I have reproduced it. I am going to update the PR